test/integration/visitor_cannot_view_private_data_test.rb

```diff
@@ -0,0 +1,32 @@
+class AuthenticatedUserPrivacyTest < ActionDispatch::IntegrationTest
+  test "visitor cannot see user data" do
+    visit '/users'
+    assert page.has_content?("Oops!")
+
+    visit '/user/1'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.